hikari/impl/shard.py

```diff
@@ -221,16 +228,18 @@ async def _receive_and_check_complete_zlib_package(
     ) -> str:
         buff = bytearray(initial_data)
 
-        while True:
+        while not buff.endswith(_ZLIB_SUFFIX):
             message = await self.receive(timeout)
 
-            if message.type != aiohttp.WSMsgType.BINARY:
-                raise errors.GatewayError(f"Unexpected message type received {message.type.name}, expected BINARY")
+            if message.type == aiohttp.WSMsgType.BINARY:
+                buff.extend(message.data)
+                continue
```

Comment:
Having it in this order seems to be more innefficient, this way is prefered

```py
            if message.type != aiohttp.WSMsgType.BINARY:
                # Raise error

            buff.extend(message.data)
```

My (not researched, just tested) guess is that the continue adds one more instruction that affects how fast this runs.

*Not a strict suggestion per say, just a comment :)*

Reply:
I was just pointing out that, for some reason, in cpython 3.8 its faster the other way around. Just a little comment I wanted to leave. Idm :)